client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/index.tsx

```diff
@@ -33,25 +35,48 @@ const Launchpad: Step = ( { navigation, flow }: LaunchpadProps ) => {
 	const verifiedParam = useQuery().get( 'verified' );
 	const site = useSite();
 	const {
-		data: { launchpad_screen: launchpadScreenOption },
+		data: { launchpad_screen: launchpadScreenOption, checklist_statuses, site_intent },
 	} = useLaunchpad( siteSlug );
+	const isSiteLaunched = site?.launch_status === 'launched' || false;
 	const recordSignupComplete = useRecordSignupComplete( flow );
 	const dispatch = useDispatch();
+	const { saveSiteSettings } = useWPDispatch( SITE_STORE );
 	const isLoggedIn = useSelector( isUserLoggedIn );
 
 	const fetchingSiteError = useSelect(
 		( select ) => ( select( SITE_STORE ) as SiteSelect ).getFetchingSiteError(),
 		[]
 	);
 
-	if ( ! isLoggedIn ) {
-		window.location.replace( `/home/${ siteSlug }` );
+	if (
+		! isLoggedIn ||
+		launchpadScreenOption === 'off' ||
+		( launchpadScreenOption === false && 'videopress' !== flow )
+	) {
+		redirectToSiteHome( siteSlug, flow );
+	}
+
+	if (
+		areLaunchpadTasksCompleted(
+			site_intent,
+			launchpadFlowTasks,
+			checklist_statuses,
+			isSiteLaunched
+		)
+	) {
+		saveSiteSettings( site?.ID, { launchpad_screen: 'off' } );
+		redirectToSiteHome( siteSlug, flow );
 	}
 
 	if ( ! siteSlug || fetchingSiteError?.error ) {
 		window.location.replace( '/home' );
 	}
 
+	function redirectToSiteHome( siteSlug: string | null, flow: string | null ) {
+		recordTracksEvent( 'calypso_launchpad_redirect_to_home', { flow: flow } );
+		window.location.replace( `/home/${ siteSlug }` );
+	}
+
```

Comment:
I've added redirectToSiteHome() to reduce duplication and ensure consistency. We hadn't been logging the calypso_launchpad_redirect_to_home event when redirecting on line 47 for the isLoggedIn check.